src/main/scala/shine/C/FunctionExecutor.scala

```diff
@@ -0,0 +1,469 @@
+package shine.C
+
+import arithexpr.arithmetic._
+import shine.C.AST.ParamDecl
+import shine.C.ParamMetaData
+import shine.DPIA.Types._
+import shine.DPIA._
+import shine.OpenCL
+import shine.OpenCL._
+import util.Time.ms
+import util.{Time, TimeSpan}
+import util.gen
+import yacx.Executor.BenchmarkResult
+import yacx.{ByteArg, Devices, DoubleArg, Executor, FloatArg, HalfArg, IntArg, KernelArg, LongArg, Options, Program, ShortArg}
+
+import scala.collection.immutable.List
+import scala.collection.Seq
+import scala.language.implicitConversions
+import scala.util.{Failure, Success, Try}
+
+//TODO: refactor
+object FunctionExecutor {
```

Comment:
why is a `C.FunctionExecutor` dealing with a kernel module?

Reply:
The CUDA-Executor can also execute C-Code. So i thought its a good idea to insert some function executor to execute and test generated c code. This should be refatored (very similar to the execution of CUDA-Code) with the CUDA-KernelExecutor.